packages/material-ui-lab/src/TreeNode/TreeNode.test.js

```diff
@@ -0,0 +1,555 @@
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('ul[role="group"]')).to.be.ok;
+    });
+
+    it('should have the attribute `aria-expanded=false` if collapsed', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded=false]')).to.be.ok;
+    });
+
+    it('should have the attribute `aria-expanded=true` if expanded', () => {
+      const { container } = render(
+        <TreeView expanded={['test']}>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded=true]')).to.be.ok;
+    });
+
+    it('should not have the attribute `aria-expanded` if no children are present', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test" />
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded]')).to.not.be.ok;
+    });
+
+    describe('when a tree receives focus', () => {
+      it('should focus the first node if none of the nodes are selected before the tree receives focus', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="1" label="one" data-testid="one" />
+            <TreeNode nodeId="2" label="two" />
+            <TreeNode nodeId="3" label="three" />
+          </TreeView>,
+        );
+
+        fireEvent.focus(getByText('one'));
+        fireEvent.focus(getByText('two'));
+        fireEvent.focus(getByText('three'));
+
+        expect(getByTestId('one')).to.be.focused;
+      });
+
+      it('should focus the selected node if a node is selected before the tree receives focus', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="1" label="one" />
+            <TreeNode nodeId="2" label="two" data-testid="two" />
+            <TreeNode nodeId="3" label="three" />
+          </TreeView>,
+        );
+
+        fireEvent.click(getByText('two'));
+
+        fireEvent.focus(getByText('one'));
+        fireEvent.focus(getByText('two'));
+        fireEvent.focus(getByText('three'));
+
+        expect(getByTestId('two')).to.be.focused;
+      });
+    });
+
+    describe('right arrow interaction', () => {
+      it('should open the node and not move the focus if focus is on a closed node', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="one" label="one" data-testid="one">
+              <TreeNode nodeId="two" label="two" />
+            </TreeNode>
+          </TreeView>,
+        );
+
+        expect(getByTestId('one')).to.have.attribute('aria-expanded', 'false');
```

Comment:
Is this really a generic container or does it have some aria role?

Reply:
It has the aria role `treeitem`